### PR TITLE
Remove old etcd client dependency

### PR DESCRIPTION
**Pull request by mfojtik**
Also updates master_config to use new etcd client.

Fixes: https://github.com/openshift/origin/issues/11573

### CODE REVIEW

File: pkg/cmd/server/etcd/etcd.go [hunk 1]
```diff
@@ -23,48 +20,17 @@ import (
 // connect to the etcd server and block until the server responds at least once, or return an
 // error if the server never responded.
 // TODO: switch this function to use EtcdHelper.
-func GetAndTestEtcdClient(etcdClientInfo configapi.EtcdConnectionInfo) (*etcdclient.Client, error) {
-	etcdClient, err := EtcdClient(etcdClientInfo)
+func GetAndTestEtcdClient(etcdClientInfo configapi.EtcdConnectionInfo) (newetcdclient.Client, error) {
+	etcdClient, err := MakeNewEtcdClient(etcdClientInfo)
```
Comment (mfojtik) [inline pkg/cmd/server/etcd/etcd.go]: @smarterclayton does this change seems correct to you?

**Comment by mfojtik** [comment]
@smarterclayton @timothysc PTAL

**Comment by mfojtik** [comment]
[test]

**Comment by soltysh** [comment]
The vendor commit change needs to be renamed to  UPSTREAM: <drop>: or UPSTREAM: coreos/go-etcd: drop.

File: pkg/cmd/server/etcd/etcd.go [hunk 2]
```diff
@@ -23,48 +20,17 @@ import (
 // connect to the etcd server and block until the server responds at least once, or return an
 // error if the server never responded.
 // TODO: switch this function to use EtcdHelper.
```
Comment (soltysh) [inline pkg/cmd/server/etcd/etcd.go]: Is this TODO still valid?

Comment (timothysc) [inline pkg/cmd/server/etcd/etcd.go]: don't think so.

**Comment by timothysc** [comment]
for part 1 removal of old, lgtm.  

You'll need to abstract the client itself on another iteration, b/c that's the v2 client.

File: pkg/cmd/server/etcd/etcd.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/server/etcd/etcd.go]: Rename newetcdclient to just `etcdclient`.

Comment (mfojtik) [inline pkg/cmd/server/etcd/etcd.go]: renamed.

**Comment by mfojtik** [comment]
@timothysc @smarterclayton @soltysh updated, PTAL.

**Comment by soltysh** [comment]
Still LGTM, but you need to update your commit message to match one of [these](https://github.com/openshift/origin/blob/c3fc160805f0c06f656deda0524046a7d9963d9e/tools/rebasehelpers/commitchecker/validate.go#L32-L37).

**Comment by timothysc** [comment]
lgtm

**Comment by smarterclayton** [comment]
You missed a spot :)

**Comment by mfojtik** [comment]
@smarterclayton hrm I had to rewrite the etcd dumper :-) please take a look if I did it right...

**Comment by smarterclayton** [comment]
Looks fine.

**Comment by soltysh** [comment]
It looks like you forgot to update tests, both integration and extended with your changes.

**Comment by mfojtik** [comment]
@soltysh I don't, this is go 1.7 (on my dev) vs. go 1.6 (in CI) :-)

**Comment by soltysh** [comment]
> @soltysh I don't, this is go 1.7 (on my dev) vs. go 1.6 (in CI) :-)

Oh yeah, that happens...

**Comment by soltysh** [comment]
This was hopefully last hit of #11661.

re-[test]

**Comment by soltysh** [comment]
Flake #11649.

**Comment by soltysh** [comment]
re-[test]

**Comment by soltysh** [comment]
Flakes: https://github.com/openshift/origin/issues/11682, https://github.com/openshift/origin/issues/10773 and #11016 

re-[test]

**Comment by soltysh** [comment]
Flake #11682 #11016 

re-[test]

**Comment by mfojtik** [comment]
flake: https://github.com/openshift/origin/issues/11799

[test]

**Comment by mfojtik** [comment]
tests are green, shipping!

[merge]

**Pull request merged**